src/renderer/MainContent.tsx

```diff
@@ -25,73 +22,35 @@ const MainContent = () => {
         const sliderCount = await getEqualizerSliderCount();
         const newIndices = Array(sliderCount)
           .fill(0)
-          .map((_, i) => i + 1);
+          .map((_, i) => i);
         setSliderIndicies(newIndices);
       } catch (e) {
-        setPeaceError(e as ErrorDescription);
+        setGlobalError(e as ErrorDescription);
       }
     };
-    if (!peaceError) {
-      fetchResults();
-    }
-  }, [peaceError, setPeaceError]);
-
-  const retryHelper = async (attempts: number, f: () => any) => {
-    for (let i = 0; i < attempts; i += 1) {
-      try {
-        await f();
-        return;
-      } catch (e) {
-        if (i === attempts) {
-          setPeaceError(e as ErrorDescription);
-          return;
-        }
-        await new Promise((resolve) => {
-          setTimeout(resolve, 500);
-        });
-      }
-    }
-  };
+    fetchResults();
+  }, [globalError, setGlobalError]);
 
   const onAddEqualizerSlider = async () => {
     try {
       await addEqualizerSlider();
-    } catch (e) {
-      setPeaceError(e as ErrorDescription);
-      return;
-    }
-    const addSlider = async () => {
-      await getProgramState();
-      await showPeaceWindow();
       const newIndices = [...sliderIndicies];
-      newIndices.push(sliderIndicies.length + 1);
+      newIndices.push(sliderIndicies.length);
       setSliderIndicies(newIndices);
-    };
-
-    retryHelper(5, addSlider);
-
-    try {
-      await closePeaceWindow();
-      // Ignore if we can't close peace window
-      // eslint-disable-next-line no-empty
-    } catch (e) {}
+    } catch (e) {
+      setGlobalError(e as ErrorDescription);
+    }
   };
 
   const onRemoveEqualizerSlider = async () => {
     try {
-      await removeEqualizerSlider();
-    } catch (e) {
-      setPeaceError(e as ErrorDescription);
-      return;
-    }
-
-    const removeSlider = async () => {
-      await getProgramState();
+      await removeEqualizerSlider(sliderIndicies.length - 1);
```

Comment:
I made the `removeEqualizerSlider` function take in the index of the slider to be removed so it's more flexible and extendible to future changes. Along with the zero-index change, `sliderIndices.length - 1`  is the correct number to pass to remove the last slider.